lib/install/management/appliance.go

```diff
@@ -639,21 +639,49 @@ func (d *Dispatcher) createAppliance(conf *config.VirtualContainerHostConfigSpec
 }
 
 func (d *Dispatcher) encodeConfig(conf *config.VirtualContainerHostConfigSpec) (map[string]string, error) {
-	if d.secret == nil {
-		log.Debug("generating new config secret key")
+	log.Debug("generating new config secret key")
+
+	s, err := extraconfig.NewSecretKey()
+	if err != nil {
+		return nil, err
+	}
+	d.secret = s
+
+	cfg := make(map[string]string)
+	extraconfig.Encode(d.secret.Sink(extraconfig.MapSink(cfg)), conf)
+	return cfg, nil
+}
 
-		s, err := extraconfig.NewSecretKey()
+func (d *Dispatcher) decryptVCHConfig(vm *vm.VirtualMachine, cfg map[string]string) (*config.VirtualContainerHostConfigSpec, error) {
+	defer trace.End(trace.Begin(""))
+
+	if d.secret == nil {
+		name, err := vm.Name(d.ctx)
+		if err != nil {
+			err = errors.Errorf("Failed to get vm name %q: %s", vm.Reference(), err)
+			return nil, err
+		}
+		// set session datastore to where the VM is running
+		ds, err := d.getImageDatastore(vm, nil, true)
+		if err != nil {
+			err = errors.Errorf("Failure finding image store from VCH VM %q: %s", name, err.Error())
```

Comment:
we planed to use errors sometime ago, but now both of them are used. I'm not going to touch that in this PR.